featurebyte/service/entity_relationship_extractor.py

```diff
@@ -17,13 +18,154 @@
 @dataclass
 class EntityRelationshipData:
     """
-    Parent data
+    Entity relationship data
     """
 
     entity_id: ObjectId
     relationship_id: ObjectId
 
 
+@dataclass
+class ServingEntityEnumeration:
+    """
+    Serving entity enumeration class is used to enumerate
+    all serving entity IDs from the given entity IDs
+    """
+
+    entity_id_to_ancestor_ids: Dict[ObjectId, Set[ObjectId]]
+    entity_id_to_descendant_ids: Dict[ObjectId, Set[ObjectId]]
+
+    @classmethod
+    def _depth_first_search(
+        cls,
+        ancestors_or_descendants_map: Dict[ObjectId, Set[ObjectId]],
+        relation_map: Dict[ObjectId, Set[ObjectId]],
+        entity_id: ObjectId,
+        ancestor_or_descendant_ids: Set[ObjectId],
+        depth: int,
+        max_depth: int,
+    ) -> None:
+        if depth > max_depth:
+            return
+
+        ancestors_or_descendants_map[entity_id].update(ancestor_or_descendant_ids)
+        for next_entity_id in relation_map[entity_id]:
+            cls._depth_first_search(
+                ancestors_or_descendants_map=ancestors_or_descendants_map,
+                relation_map=relation_map,
+                entity_id=next_entity_id,
+                ancestor_or_descendant_ids=ancestor_or_descendant_ids | {entity_id},
+                depth=depth + 1,
+                max_depth=max_depth,
+            )
+
+    @classmethod
+    def create(
+        cls, relationships_info: List[EntityRelationshipInfo], max_depth: int = 5
+    ) -> "ServingEntityEnumeration":
+        """
+        Create entity relationship graph from relationships info
```

Comment:
Typo `entity relationship graph` here and in the return type.

Reply:
Good catch, will fix it in the following PR.